examples/cpp/hello_world/README.md

```diff
@@ -0,0 +1,184 @@
+
+```shell
+user@machine:example_path$ ./hello_world publisher
+Publisher running. Please press Ctrl+C to stop the Publisher at any time.
+```
+
+#### Windows
+
+```powershell
+example_path> hello_world.exe publisher
+Publisher running. Please press Ctrl+C to stop the Publisher at any time.
+```
+
+### Hello world subscriber
+
+#### Ubuntu ( / MacOS )
+
+```shell
+user@machine:example_path$ ./hello_world subscriber
+Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
+```
+
+#### Windows
+
+```powershell
+example_path> hello_world.exe subscriber
+Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
+```
+
+## Expected output
+
+It does not matter which entity is launched first, because the publisher has a condition to wait until the first match to start sending hello world samples.
+The expected output on both cases is a first displayed message acknowledging they have matched, followed by the amount of samples sent or received until user press Ctrl+C.
+
+### Hello world publisher
+
+```shell
+Publisher running. Please press Ctrl+C to stop the Publisher at any time.
+Publisher matched.
+Message: 'Hello world' with index: '1' SENT
+Message: 'Hello world' with index: '2' SENT
+Message: 'Hello world' with index: '3' SENT
+...
+```
+
+### Hello world subscriber
+
+```shell
+Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
+Subscriber matched.
+Message: 'Hello world' with index: '1' RECEIVED
+Message: 'Hello world' with index: '2' RECEIVED
+Message: 'Hello world' with index: '3' RECEIVED
+...
+```
+
+When pressed Ctrl+C to stop one of the applications, the other will detect the unmatched status, and will display an informative message.
+This output example represents stopping the subscriber application.
+
+### Hello world publisher
+
+```shell
+...
+Message: 'Hello world' with index: '9' SENT
+Message: 'Hello world' with index: '10' SENT
+Message: 'Hello world' with index: '11' SENT
+Publisher unmatched.
+Message: 'Hello world' with index: '12' SENT
+Message: 'Hello world' with index: '13' SENT
+...
+```
+
+### Hello world subscriber
+
+```shell
+...
+Message: 'Hello world' with index: '9' RECEIVED
+Message: 'Hello world' with index: '10' RECEIVED
+Message: 'Hello world' with index: '11' RECEIVED
+user@machine:example_path$ |
+```
+
+Stopping the subscriber application does not make the publisher stop sending data. But in the opposite case, stopping the publisher application will make subscriber stop receiving data.
+
+### Hello world publisher
+
+```shell
+...
+Message: 'Hello world' with index: '9' SENT
+Message: 'Hello world' with index: '10' SENT
+Message: 'Hello world' with index: '11' SENT
+user@machine:example_path$ |
+```
+
+### Hello world subscriber
+
+```shell
+...
+Message: 'Hello world' with index: '9' RECEIVED
+Message: 'Hello world' with index: '10' RECEIVED
+Message: 'Hello world' with index: '11' RECEIVED
+Subscriber unmatched.
+```
+
+## Advanced configuration
```

Comment:
I don't think we need this title. You'd need to adjust the level of the rest of titles
```suggestion
```